shared/validate/validate.go

```diff
@@ -763,3 +763,46 @@ func IsAbsFilePath(value string) error {
 
 	return nil
 }
+
+func ParseNetworkVLANList(rawVLANValues []string) ([]string, error) {
+	var networkVLANList []string
+	for _, vlan := range rawVLANValues {
+		if err := IsNetworkVLAN(vlan); err != nil {
+			networkVLANList = append(networkVLANList, vlan)
+		} else if err := IsNetworkVLANRange(vlan); err != nil {
```

Comment:
Please can you make `err = ` assignments on their own lines rather than using the `err := IsNetworkVLANRange(vlan); err != nil` format as we prefer this to avoid confusion around err scopes.

Also, this looks quite inefficient, as you're calling `IsNetworkVLANRange` which calls `ParseNetworkVLANRange` internally, and then if it passes validation, it then calls `ParseNetworkVLANRange` again to get the actual ranges.

Why not just call `ParseNetworkVLANRange` directly and check its error value?

Reply:
Got it - I was following the of ParseUnit32Range and thought it was the preferred way of doing things.

Reply:
ParseUint32Range doesn't call IsUint32Range though.

Reply:
No, but IsUint32Range calls ParseUInt23Range first to catch the error, and then proceeds to call ParseUint32Range again.

```
// IsUint32Range validates whether the string is a uint32 range in the form "number" or "start-end".
func IsUint32Range(value string) error {
	_, _, err := ParseUint32Range(value)
	return err
}
```

Reply:
I am a similar thing here - use `IsNetworkVLANRange` to call the `ParseNetworkVLANRange` for validation, and then use `ParseNetworkVLANRange` to get the range boundaries. I was confused by `ParseUint32Range`'s approach as well at first, but I think it does provide more readability?

Reply:
The `IsUint32Range` function is OK as it just calls `ParseUint32Range` without returning the parsed value. Its use is during the validation stage, before it is used. I agree that it is still somewhat inefficient to have to call `ParseUint32Range` later in device start up to actually access the parsed native values. But that is different to having `ParseNetworkVLANList` call `IsNetworkVLANRange` which calls `ParseNetworkVLANRange`, only to have it call `ParseNetworkVLANRange` again within the same function on the next line.

Reply:
Got it - it makes more sense now.